rfcs/000-search-revamp.md

```diff
@@ -0,0 +1,124 @@
+- Feature Name: Search Revamp
+- Start Date: 2021-09-20
+- RFC PR: todo
+
+# Search Revamp
+
+## Summary
+
+Revamp search with a cleaner backend implementation, more robust queries, better customizability, and
+making use of modern v7 ES features.  
+
+## Motivation
+
+At Lyft, search enhancement are the #1 most frequent Amundsen feedback. This is no surprise, as the current 
+technical state of search has not changed much from MVP.
+
+We see search as serving two distinct (but related) use cases:
+  - known-data searches, which have a starting point (e.g. tables with a `user_id` column in the `production` schema) 
+    and are more interested in listing the available data matching a known condition
+  - discovery searches, which are looking to find previously-unknown data that would be relevant for solving
+    one's problem
+
+Although we believe Amundsen has been particularly lacking in the discovery search category, our proposal is seeking
+to improve the experience of both search types. 
+
+
+## UI/UX Details
+
+Although the bulk of proposed changes will be to search result quality and relevance, we're also proposing
+a handful of UI/UX changes to filters. We're currently working on designs and will share those as soon as they're
+available.
+
+#### Multiple Values Per Filter.
+Allow multiple values (likely comma-separated) for filters, which has been very highly requested. 
+
+There is an obvious question of whether multiple values would be AND'ed or OR'ed together. Our goal is to provide
+additional expressiveness without going down the path of supporting full logical expressions. To better understand the problem,
+keep in mind that there's two kind of fields: single-valued fields (table name, schema name) and multi-valued fields (columns, tags, badges).
+Our proposed tradeoff between expressiveness and simplicity is to combine filter values for single-valued 
+fields with OR (e.g. "staging,production" filter applied to the "schema" field)
+whereas for multi-valued fields we'd give users an options to choose AND or OR 
+(a user would choose AND/OR to apply the filters "product_id,user_id" to the "column" field)
+
+#### Negation filtering
```

Comment:
We are definitely not looking back to using Atlas Search API but there are other Atlas users which might be impacted by removing search proxy. 

Arguments for it would be simplicity of Amundsen implementation and aligned search experience but it might hurt the simplicity of Amundsen and Atlas deployment if some users are satisfied with Atlas search and don't want to go into trouble of setting up ES & running databuilder search sync.

Reply:
I am a million percent in favor of removing. I'll run some informal polling first, of course, but I think the burden of maintaining it probably far exceeds the combined user burden (if any) of spinning up elasticsearch. 

This is not to mention that the current Atlas proxy supports only a fraction of what's supported by the ES proxy. No user/dashboard/feature search, for starters.

Reply:
I think we are fine removing it as long as we provide some transition period so users have comfortable time for switching into new approach. We'd promote the approach that is already outlined in docs (using databuilder search extractor).